test/test_example_parsers/test_dice_roll_parser.py

```diff
@@ -0,0 +1,65 @@
+import pytest
+
+from plusminus.examples.example_parsers import DiceRollParser
+
+
+@pytest.fixture
+def die():
+    return DiceRollParser()
+
+
+class TestSingleROll:
+    def test_single_roll_max_one(self, die):
+        assert die.evaluate('d1') == 1
+
+    @pytest.mark.parametrize('die_max',
+                             [2, 4, 5, 10, 100, ],
+                             )
+    def test_single_roll_ranges(self, die,
+                                die_max):
+        assert 1 <= die.evaluate(f'd{die_max}') <= die_max
+
+    def test_single_roll_error_empty_range(self, die):
+        with pytest.raises(ValueError):
+            die.evaluate('d0')
+
+
+class TestMultipleRolls:
+    @pytest.mark.parametrize('die_max',
+                             [2, 3, 4, 5, 10, 100, ]
+                             )
+    def test_zero_rolls_returns_zero(self, die,
+                                     die_max):
+        assert die.evaluate(f'0d{die_max}') == 0
+
+    @pytest.mark.parametrize('num_rolls',
+                             [1, 2, 4, 5, 6, 10, 100, ],
+                             )
+    def test_multiple_rolls_max_one(self, die,
+                                    num_rolls):
+        assert num_rolls <= die.evaluate(f'{num_rolls}d1') == num_rolls
+
+    @pytest.mark.parametrize('num_rolls',
+                             [1, 2, 4, 5, 10, 100, ],
+                             )
+    @pytest.mark.parametrize('die_max',
+                             [2, 4, 5, 10, 100, ]
+                             )
+    def test_single_roll_ranges(self, die,
+                                num_rolls, die_max):
+        assert num_rolls <= die.evaluate(f'{num_rolls}d{die_max}') <= num_rolls*die_max
```

Comment:
I don't know how useful this test and the following class are for this parser, unless one repeats the tests enough times to at least have a good chance of producing an edge or out of range result. The tests do confirm that the parser evaluates the input successfully, and that it is within the range of expected outputs, which is marginally more than the [original set of tests](https://github.com/pyparsing/plusminus/blob/1cd60b8f802e9b77c166cf5339437612b6647bde/plusminus/examples/example_parsers.py#L125-L127).